lib/helpers/broadcast.js

```diff
@@ -36,12 +36,27 @@ const broadcastTransformer = (topic) => {
  * @param {String} id
  * @return {Promise}
  */
+const fetchers = {};
 async function fetchById(id) {
+  // If we have an in-flight request, return existing promise:
+  if (fetchers[id]) {
+    logger.debug('Using pending request for broadcast', { id });
+    return fetchers[id];
+  }
+
+  // Otherwise, start a new request, keeping a record of it
+  // in case we fetch this same ID again before it resolves:
   logger.debug('Fetching broadcast', { id });
+  fetchers[id] = graphql.fetchBroadcastById(id);
+
+  const broadcast = broadcastTransformer(await fetchers[id]);
 
-  const broadcast = broadcastTransformer(await graphql.fetchBroadcastById(id));
+  // Once we've finished the request, store the result
+  // in cache and clear our "pending request" record:
+  helpers.cache.topics.set(id, broadcast);
```

Comment:
Should this be `broadcasts` instead of `topics?
```suggestion
  helpers.cache.broadcasts.set(id, broadcast);
```

Reply:
Eep, yes!